src/models/model.cpp

```diff
@@ -1256,7 +1256,9 @@ void model::do_batch_begin_cbs(execution_mode mode) {
       break;
     case execution_mode::validation:
     case execution_mode::testing:
-      cb->on_batch_evaluate_begin(this);
+      if (get_step() % cb->get_batch_interval() == 0) {
```

Comment:
I agree with making this the default behavior, but I worry it might introduce bugs in our callbacks. We should make this change in a separate PR.

Reply:
Will make the a separate PR, may be it is better implemented in the derived rather than base class.

Reply:
I think we have two good options:
- Keep `m_batch_interval` in the base class and make it a private variable. (This is my preference).
- Remove `m_batch_interval` from the base class and reimplement it in every derived class that needs it.